cmp/clock/Clock.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+
+import {box, span} from '@xh/hoist/cmp/layout';
+import {hoistCmp, HoistModel, managed, useLocalModel, XH} from '@xh/hoist/core';
+import {action, observable} from '@xh/hoist/mobx';
+import {MINUTES, ONE_SECOND} from '@xh/hoist/utils/datetime';
+import {Timer} from '@xh/hoist/utils/async';
+import {isEmpty, isNil, isNumber} from 'lodash';
+import moment from 'moment';
+import PT from 'prop-types';
+
+/**
+ * A component to display the current time.
+ *
+ * Automatically updates on a regular interval to stay current.
+ */
+export const [Clock, clock] = hoistCmp.withFactory({
+    displayName: 'Clock',
+    className: 'xh-clock',
+
+    render({timezone, format, updateInterval, prefix, suffix, ...props}) {
+        const impl = useLocalModel(LocalModel);
+        impl.setData({timezone, format, updateInterval, prefix, suffix});
+
+        return box({
+            ...props,
+            item: span(impl.display)
+        });
+    }
+});
+Clock.propTypes = {
+    /**
+     * Timezone id. The user's local system time will be used if omitted.
+     * @see {@link https://docs.oracle.com/middleware/12212/wcs/tag-ref/MISC/TimeZones.html} */
+    timezone: PT.string,
+
+    /** moment.js date format string. @see {@link https://momentjs.com/docs/#/displaying/format/} */
+    format: PT.string,
+
+    /** Prefix for the displayed text */
+    prefix: PT.string,
+
+    /** Suffix for the displayed text */
+    suffix: PT.string,
+
+    /** Frequency (ms) for updating the displayed text */
+    updateInterval: PT.number
+};
+
+@HoistModel
+class LocalModel {
+    timezone;
+    format;
+    updateInterval;
+    prefix;
+    suffix;
+
+    offset;
+    offsetException;
+
+    @observable display = '';
+
+    @managed
+    timer = Timer.create({
+        runFn: () => this.refreshDisplay(),
+        interval: ONE_SECOND
```

Comment:
Should we create this lazily on the first call to `setData()` when we know the actual interval desired vs. also having `ONE_SECOND` here?